config/env_sample.hjson

```diff
@@ -252,7 +253,36 @@
                 and membership.id is not null 
                 and REGEXP_EXTRACT(`group`.id, r'.*:(.*)') IN UNNEST(@course_ids)
                 '''
-        }
+        },
+        "canvas_pages": {
+            "query":
+                '''
+                SELECT 'canvas_pages' AS resource_type,
+                CONCAT(CAST(REGEXP_EXTRACT(`group`.id, r'.*:(.*)') AS INT64) - @canvas_data_id_increment, 
```

Comment:
Interesting solution! This query would fail if it can't do the CAST though, I'm not sure the likelihood of that. 

You *could* use this pattern in these queries to make it safer and always return 0. Perhaps we want the cron to error out though. Maybe something to keep in mind if we do see a failure here. 

```
IFNULL(SAFE_CAST(REGEXP_EXTRACT(`group`.id, r'.*:(.*)') AS INT64), @canvas_data_id_increment) - @canvas_data_id_increment, 
```

Reply:
added IFNULL(SAFE_CAST()) pattern into query.